src/AzureAuth.Test/PublicClientAuthTest.cs

```diff
@@ -0,0 +1,115 @@
+
+            Action nullTelemetryService = () => new PublicClientAuth(this.logger, this.env.Object, null, null);
+            nullTelemetryService.Should().Throw<ArgumentNullException>().WithParameterName("telemetryService");
+
+            Action nullTokenFetcher = () => new PublicClientAuth(this.logger, this.env.Object, this.telemetryService.Object, null);
+            nullTokenFetcher.Should().Throw<ArgumentNullException>().WithParameterName("tokenFetcher");
+        }
+
+        [Test]
+        public void Good_Auth_Returns_TokenResult()
+        {
+            // Arrange
+            var correlationId = new Guid("6ed5394e-511d-4a45-b41d-f949bf7ec523");
+            var authFlowName = "TestAuthFlow";
+            var expected = new TokenResult(new JsonWebToken(Fake.Token), correlationId);
+            var tokenFetcherResult = new MsalWrapper.Result()
+            {
+                Attempts = new List<AuthFlowResult>() { new AuthFlowResult(expected, Array.Empty<Exception>(), authFlowName) },
+            };
+
+            // We Expect TokenFetcher to be called with specific transformations on the arguments we are giving the subject.
+            // * The AuthMode should be combined into 1 bit flag.
+            // * The Prompt should be prefixed.
+            var expectedMode = AuthMode.Web | AuthMode.DeviceCode; // These 2 auth modes chosen because they are x-plat.
+            var expectedPrompt = $"AzureAuth: {this.prompt}";
+
+            this.tokenFetcher.SetupSequence(
+                t => t.AccessToken(
+                    this.logger, this.client, this.tenant, this.scopes, expectedMode, this.domain, expectedPrompt, this.timeout))
+                .Returns(tokenFetcherResult);
+
+            // The AuthMode should be Combined, and run through the extension to disable interacive auth if needed.
+            this.env.Setup(e => e.Get("AZUREAUTH_NO_USER")).Returns((string)null);
+            this.env.Setup(e => e.Get("Corext_NonInteractive")).Returns((string)null);
+
+            // One AuthFlow Telemetry event should be sent.
+            // We don't need to assert the details of those events here because they are unit tested
+            // separately in AuthFlowResultExtensionsTest for converting an AuthFlowResult to EventData.
+            this.telemetryService.SetupSequence(t => t.SendEvent("authflow_TestAuthFlow", It.IsAny<EventData>()));
+
+            // Act
+            TokenResult subject = this.Subject().Token(this.client, this.tenant, this.scopes, new[] { AuthMode.Web, AuthMode.DeviceCode }, this.domain, this.prompt, this.timeout);
+
+            // Assert
+            subject.Should().Be(expected);
+            // These logs are only those generated by this class/test.
+            // What we do not see here are the debug and trace logs generated by the services that are mocked for the test.
+            this.logTarget.Logs.Should().BeEmpty();

```

Comment:
Is this essentially the same thing as `logMock.VerifyNoOtherCalls();`

Reply:
Partially. It's like doing that + no mocking any calls to the logger at all.